pvlib/irradiance.py

```diff
@@ -2915,3 +2917,61 @@ def dni(ghi, dhi, zenith, clearsky_dni=None, clearsky_tolerance=1.1,
             (zenith < zenith_threshold_for_zero_dni) &
             (dni > max_dni)] = max_dni
     return dni
+
+
+def component_sum_irradiance(zenith,
+                             ghi=None,
+                             dhi=None,
+                             dni=None,
+                             clearsky_dni=None):
+    """
+    Use the component sum equations to calculate the missing series, using
+    the other available time series. One of the three parameters (ghi, dhi,
+    dni) is passed as None, and the other associated series passed are used to
+    calculate the missing series value.
+
+    Parameters
+    ----------
+    zenith : Series
+        True (not refraction-corrected) zenith angles in decimal
+        degrees, with datetime index. Angles must be >=0 and <=180. Must have
+        the same datetime index as ghi, dhi, and dni series, when available.
+    ghi : Series, default None
+        Pandas series of dni data, with datetime index. Must have the same
+        datetime index as dni, dhi, and zenith series, when available.
+    dhi : Series, default None
+        Pandas series of dni data, with datetime index. Must have the same
+        datetime index as ghi, dni, and zenith series, when available.
+    dni : Series, default None
+        Pandas series of dni data, with datetime index. Must have the same
+        datetime index as ghi, dhi, and zenith series, when available.
+    clearsky_dni : Series, (optional, default None)
+        Pandas series of clearsky dni data, calculated via the
+        get_clearsky function. Must have the same datetime index as ghi, dhi,
+        dni, and zenith series, when available. This is an optional field that
+        is only used when calculating the DNI component sum series.
+
+    Returns
+    -------
+    component_sum_df : Dataframe
+        Pandas series of 'ghi', 'dhi', and 'dni' columns with datetime index
+    """
+    if (ghi, dhi) is not None and dni is None:
+        dni = pvlib.irradiance.dni(ghi, dhi, zenith,
+                                   clearsky_dni=clearsky_dni,
+                                   clearsky_tolerance=1.1)
+    elif (dhi, dni) is not None and ghi is None:
+        ghi = (dhi + dni * tools.cosd(zenith))
+    elif (dni, ghi) is not None and dhi is None:
+        dhi = (ghi - dni * tools.cosd(zenith))
+    else:
+        wrn_txt = ("No component sum calculated. Please recheck \n"
+                   "passed ghi, dni, and dhi parameters to check \n"
+                   "exactly one field out of the three is set to None.")
+        warnings.warn(wrn_txt, UserWarning)
+    # Merge the outputs into a master dataframe containing 'ghi', 'dhi',
+    # and 'dni' columns
+    component_sum_df = pd.concat({'ghi': ghi,
+                                  'dhi': dhi,
+                                  'dni': dni}, axis=1)
```

Comment:
```suggestion
    component_sum_df = pd.DataFrame({'ghi': ghi, 'dhi': dhi, 'dni': dni})
```

perhaps a bit more idiomatic